src/adr/0041-james-servers-offering.md

```diff
@@ -0,0 +1,159 @@
+can make new comers confused. It currently satisfies operator looking for self hosting solutions as well as 
+those looking for an extensible server.
+
+ - **JPA SMTP** (`server/container/guice/jpa-smtp`) demonstrates a Mail Transfer Agent, dedicated to mail
+processing, that does not ship the James Mailbox, as well as Mail Delivery protocols like IMAP, JMAP or POP3.
+
+We can notice that this offer is specified in technical terms, and does not directly map to identified user needs.
+
+Also the large amount of servers makes the choice harder to make for newcomers.
+
+## Decision
+
+We should deliver a subset of mail server directly addressing users needs, and brand them according to their use cases
+and not due to their technical details.
+
+In order to get a clearer offer we should first reduce the server cardinality. This can be achieved via:
+
+ - **Spring server** deprecation and removal
+ - Merging Guice servers with their LDAP variations (see [ADR 36 about guice module choosing](0036-against-use-of-conditional-statements-in-guice-modules.md))
+
+We should then rename our servers in order to match the use cases:
+
+ - **Basic server** addresses the self-hosting use case. It will be based on current *jpa-guice* server but with the clear intent to 
+lower adoption barriers. It exposes limited configurations in order not to confuse the user. Configuration and documentation is centered 
+around self hosting use cases. Configuration and documentation addresses the self-hosting usage without exposing underlying James concepts.
+
+The **Basic Server** is a subset of the advanced server, and differs from it from documentation and configuration. Users of the basic server feeling
+constraint by its minimalistic configuration capabilities and features would then be able to easily upgrade to the advanced server.
+
+ - **Advanced server** addresses the customization use cases. It will be based on the current *jpa-guice* server. It differs from the basic server
+as more features will be packaged as part of it, configuration shows via examples advanced usage possibilities, and the documentation is centered 
+around feature extensibility.
+
+ - **Cassandra Guice** (TODO rebrand me) still have community traction, 
+
+Note that transitionning from basic server to the advanced one will be easy as they rely on the same technologies.
+
+ - **Distributed server** will remain unchanged.
+
+ - **Testing server** will rely on current `memory guice` server.
+
+ - **Mail processing server** will rely on current `JPA SMTP guice` server.
+
+Also, we should separate our server application from the technical guice bindings, and group them together for better discoverability.
+
+We should adopt this folder structure:
+
+```
+server/distributions
+\_____ basic
+\_____ advanced
+\_____ distributed
+\_____ testing
+\_____ mail-processing
+\_____ cassandra-guice (TODO rebrand me please)
```

Comment:
Just a thought, couldn't that be grouped with basic and advanced servers, but, to retake a wording from @rouazana , being called a different flavor?

Like :
* basic server has 2 flavors -> JPA flavor and Cassandra flavor
* advanced server has 2 flavors -> JPA flavor and Cassandra flavor
etc

Reply:
For the Advanced Server, I think that would work just fine. My understanding is: we could say that there are "infinite" configurations possible (well, not really, but just to simplify), but that there are "common patterns" that we call "flavours".

For the Basic Server, I think it goes against the principle of what we are trying to achieve by the use case.

Reply:
I agree with the fact that Basic server should stay very simple and not mingle different flavors thanks !

So yes the Cassandra guice version could be a flavor for the advanced server, alongside the JPA flavor